ADBDEV-4726-38 Check that instrument is not null

dnskvlnk opened the pull request:
Check that instrument is not null

show_sort_info dereferences instrument without checking that it's not null.
We can't guarantee that instrument is always valid. This patch adds null-check

Commit:
Check that instrument is not null

Commit:
Check that instrument is not null

show_sort_info dereferences ((PlanState *) sortstate)->instrument without
checking that it's not null. This patch adds an assert

RekGRpth commented on src/backend/commands/explain.c:
```diff
@@ -2668,6 +2668,8 @@ show_sort_info(SortState *sortstate, ExplainState *es)
 	if (!es->analyze)
 		return;
 
+	Assert(((PlanState *) sortstate)->instrument);
```
Why did you decide that `((PlanState *) sortstate)->instrument` cannot be NULL? Maybe it would be better to return from a function, as is done a few lines below?

dnskvlnk replied:
It can't be null because it's dereferenced in the next line

Commit:
Check that instrument is not null

show_sort_info dereferences instrument without checking that it's not null.
We can't guarantee that instrument is always valid. This patch adds Insist

RekGRpth commented:
add title to PR description

dnskvlnk commented:
@RekGRpth what do you mean? It has a title like in commit

RekGRpth commented:
rebase needed

RekGRpth commented on src/backend/commands/explain.c:
```diff
@@ -2668,6 +2668,8 @@ show_sort_info(SortState *sortstate, ExplainState *es)
 	if (!es->analyze)
 		return;
 
+	Insist(((PlanState *) sortstate)->instrument);
```
see above
https://github.com/arenadata/gpdb/blob/fd5939d4f491238bbd28be88b98949cd0e2d6fe0/src/backend/commands/explain.c#L2177

dnskvlnk replied:
What do you want to say by this?

RekGRpth replied:
planstate->instrument may be NULL and it is OK

dnskvlnk replied:
Fixed

RekGRpth replied:
update PR description according to changes

dnskvlnk replied:
Done

Commit:
Change Insist to null-check

Commit:
Check that instrument is not null

show_sort_info dereferences instrument without checking that it's not null.
We can't guarantee that instrument is always valid. This patch adds null-check
